### PR TITLE
added reverse iterator

### PR DESCRIPTION
**Pull request by serk12**
http://wiki.scummvm.org/index.php/TODO#Iterator_handling

### CODE REVIEW

File: test/common/list.h [hunk 1]
```diff
@@ -86,6 +86,50 @@ class ListTestSuite : public CxxTest::TestSuite
...
+		// // Also test the postinc
+		// TS_ASSERT_EQUALS(*iter, -11);
+		// iter++;
+		// cIter++;
+		// TS_ASSERT_EQUALS(iter, container.end());
+		// TS_ASSERT_EQUALS(cIter, container.end());
+		// TS_ASSERT_EQUALS(iter, cIter);
+		//
+		// cIter = iter;
+		// TS_ASSERT_EQUALS(iter, cIter);
```
Comment (tobiatesan) [inline test/common/list.h]: Why do we have commented lines in a commit?
Why not just introducing these lines, uncommented, in a successive commit?

File: TODO [hunk 2]
```diff
@@ -2,11 +2,38 @@ We moved TODO list to our Wiki. You may find it at the following link:
```
Comment (tobiatesan) [inline TODO]: Typically git commit messages tell what the commit _does_ upon being applied, so they look like "Add hardcoded reverse iterator".

**Comment by tobiatesan** [comment]
Typically git commit messages tell what the commit does upon being applied, so they look like "Add hardcoded reverse iterator".

**Comment by criezy** [comment]
I have not looked at the changes yet, but the commit messages fo not follow our conventions. See http://wiki.scummvm.org/index.php/Commit_Guidelines

Also from a practical point of view for the future, I would recommend that you keep your master clean and do work like that in branches. This give you the ability to keep your master synchronised with our main repo without introducing merge commits.

**Comment by csnover** [comment]
@serk12 are you planning on continuing to work on this patch?

**Comment by serk12** [comment]
No, sorry. I'm with other projects now.
I thought I closed the pull request time ago. So I close it for good.

**Pull request closed without merging**